src/ast/ast.cpp

```diff
@@ -181,6 +179,7 @@ Cast::Cast(SizedType cast_type, Expression *expr, location loc)
 Tuple::Tuple(ExpressionList &&elems, location loc)
     : Expression(loc), elems(std::move(elems))
 {
+  is_literal = true;
```

Comment:
Not sure this is correct. Isn't this saying all tuples are literal? Perhaps you want to check that all fields in the tuple are literal as well? But I'm unsure what the use case is.

Reply:
It's just saying that the tuple this expression: `(1, 2, 3)` is a literal, whereas this `$a`, which was previously initialized as a tuple `$a = (1,2,3)` is not. This is used for type manipulation in semantic_analyser for concrete expressions like `@map[(1, 2)] = 1;` vs `@map[$a] = 1;`

Reply:
But my reading of this code is that something like `(1, 2 $three)` will be marked as a literal. Is there something I'm missing?

Perhaps you actually want a `dyn_cast` to check that the AST node is directly a tuple. `is_literal` is more for integers/strings

Reply:
> Perhaps you actually want a dyn_cast to check that the AST node is actually a tuple. is_literal is more for integers/strings

Ah ok. Yeah, this is better. Will change.